Ginger/GingerCoreNET/Run/RunsetExecutor.cs

```diff
@@ -922,11 +922,11 @@ public static void ClearAndResetVirtualAgents(RunSetConfig runset, GingerExecuti
                         {
                             var virtualAgent = (Agent)appAgents[i].Agent;
 
-                            var realAgent = runset.ActiveAgentList.FirstOrDefault(x => ((Agent)x).Guid.ToString() == virtualAgent.ParentGuid.ToString());
+                            var realAgent = runset.ActiveAgentListWithRunner.Select(y => y.Value).Where(x => ((Agent)x).Guid.ToString() == virtualAgent.ParentGuid.ToString()).FirstOrDefault();
```

Comment:
If the GUIDs being compared are both actual GUIDs, use the `guid.equals` or `guid.compareto` methods instead of converting them to strings and comparing them.

Reply:
Updated